agent/engine/task_manager.go

```diff
@@ -489,8 +494,27 @@ func (mtask *managedTask) cleanupTask(taskStoppedDuration time.Duration) {
 		cleanupTimeBool <- true
 		close(cleanupTimeBool)
 	}()
+	// wait for the cleanup time to elapse, signalled by cleanupTimeBool
 	for !mtask.waitEvent(cleanupTimeBool) {
 	}
+	stoppedSentBool := make(chan bool)
+	go func() {
```

Comment:
There's no `context` wired in to most of our codebase as most of it was written before `context` existed.  We could add that to the `managedTask` but that's outside the scope of my change here.

I don't think that always passing `context` is a hard rule that we should enforce.  I think that whether or not we should use `context` depends on what the code is doing.